core/tests/ts-integration/tests/fees.test.ts

```diff
@@ -207,10 +276,16 @@ testFees('Test fees', () => {
     });
 
     afterAll(async () => {
+        await testMaster.deinitialize();
+        await mainNode.killAndWaitForShutdown();
         // Returning the pubdata price to the default one
-        await setInternalL1GasPrice(alice._providerL2(), undefined, undefined, true);
 
-        await testMaster.deinitialize();
+        // Restore defaults
+        setTransactionSlots(pathToHome, fileConfig, 8192);
+        deleteInternalEnforcedL1GasPrice(pathToHome, fileConfig);
+        deleteInternalEnforcedPubdataPrice(pathToHome, fileConfig);
+        mainNode = await mainNodeSpawner.spawnMainNode();
+        await __ZKSYNC_TEST_CONTEXT_OWNER__.setPid(mainNode.proc.pid!);
```

Comment:
I couldn't find a better solution either. Our jest-setup is not flexible enough to do it any better.
`TestContextOwner` assumes that all cleanup can be done within `afterAll`, however it needs L1, L2 to stay alive during `teardownContext`. So if test replaces L2 (like by spawning a new server), then test is having a hard time letting context owner know that it has to be terminated in the end. I think the way you do it is reasonable.
I don't understand how it worked before? Was this test running under assumption that it's fine to leave a hanging server? Because, technically, if you do `mainNode.unref()` the test passes, it's just leaving server running :)

Reply:
It was working before, just because we useed `killall`